pwnlib/tubes/process.py

```diff
@@ -667,6 +667,40 @@ def kill(self):
         """
         self.close()
 
+    def terminate(self):
+        """terminate()
+
+        Terminates the process by sending SIGTERM.
+        
+        This is a more graceful way to stop a process compared to kill(),
+        which sends SIGKILL. The process has a chance to clean up and
+        exit gracefully when receiving SIGTERM.
+        
+        Examples:
+        
+            >>> p = process(['sleep', '10'])
+            >>> p.terminate()
+            >>> p.poll() is not None
+            True
+        """
+        if self.proc is None:
+            return
+            
+        # Check if process is still running
+        if self.poll() is not None:
```

Comment:
The poll check should be done afterwards to detect if the process exited immediately.